api/api.go

```diff
@@ -615,8 +615,14 @@ func createNamespaceWithOwnerID(ctx context.Context, logger *logger.Logger, name
 
 	err = graphQLclient.Run(ctx, request, &response)
 
-	if err != nil {
-		return nil, errors.Wrapf(err, "Unable to create namespace %s for ownerId %s", name, ownerID)
+	// Only fall back to native graphql errors if there are no in-band errors.
+	if err != nil && len(response.Data.CreateNamespace.Errors) == 0 {
```

Comment:
I think this type of comparison is much easier to understand when the condition list is in priority order. That way you don't need an `&&` in each case, and the cyclomatic complexity is much lower. It also removes the need for inline comments, which should not be unnecessary here. These are simple comparisons that can be made obvious by the structure of the code.

```go
switch {
case len(response.Data.CreateNamespace.Errors) > 0:
    return response.Data.CreateNamespace.Errors
case err != nil:
    return err
case len(response.Errors) > 0:
    return response.Errors
}
```

I believe this is the order that you have now, but I'm not sure why `err` has higher priority than `response.Errors`. I guess in practice it is unlikely that both of those are set.